compiler/crates/relay-transforms/src/declarative_connection.rs

```diff
@@ -273,6 +273,30 @@ impl Transformer for DeclarativeConnectionMutationTransform<'_> {
                     Some(connections_arg) => {
                         let edge_typename_arg = node_directive.arguments.named(*EDGE_TYPENAME_ARG);
                         if let Some(edge_typename_arg) = edge_typename_arg {
+                            if let Some(edge_typename_value) = edge_typename_arg
+                                .value
+                                .item
+                                .get_constant()
+                                .and_then(|c| c.get_string_literal())
```

Comment:
Out of curiosity, what happens if you try to use a variable here? Does that trigger some other reasonable error message?

Reply:
It doesn't error. Is using a variable considered an error? I've added a test case that demonstrates that it currently doesn't fail.